packages/abstractions/src/serialization/parseNode.ts

```diff
@@ -104,4 +104,9 @@ export interface ParseNode {
 	 * @return the byte array value of the node.
 	 */
 	getByteArrayValue(): ArrayBuffer | undefined;
+	/**
+	 * Gets the value for the current node
+	 * @return a parse node value for the current node element.
+	 */
+	getNodeValue(): unknown;
```

Comment:
I'm not sure if this is the right thing to do as the `parseNode` interface will no longer align with the implementation of the other languages.

Reply:
Would it make useful to define two helper methods `IsNumberNode(ParseNode)` and `IsStringNode(ParseNode)` that essentially return true or false if the intent is to reduce generated code? This may also introduce type safety benefits rather than returning `unknown` types.

https://github.com/microsoft/kiota-typescript/blob/cd04c06ef4465a29164fc7d1eb9f3cd314d41609/packages/abstractions/src/serialization/untypedNumber.ts#L22

Reply:
Okay what would the complete code look like, for generating either a string | number for example

Reply:
probably something like this? So that return values are also typed rather than returning `unknown`.

```suggestion
export function createPrimitivesFromDiscriminatorValue(parseNode: ParseNode | undefined) : number | string | undefined {
    if(IsNumberNode(parseNode)){
         return parseNode.getNumberValue();      
    }
     if(IsStringNode(parseNode)){
         return parseNode.getStringValue();      
    }
}
```

Reply:
The return value won't be unknows since the type definition has already been checked:
![nodeValue](https://github.com/microsoft/kiota-typescript/assets/1716093/bac92752-4f8f-46a8-8200-87b7c68baf1e)

Your suggestion is giving an error confirm whether there's something I missed:
![untypedNode](https://github.com/microsoft/kiota-typescript/assets/1716093/85e4c4be-b526-40f0-964f-27f49ca62b12)

Reply:
Just to confirm here, the suggestion would be to add methods like the ones I referenced. Not using them.

Reply:
I have refactored the generated code should look something like this:
``` Typescript
export function createPrimitivesFromDiscriminatorValue3(parseNode: ParseNode | undefined) : number | string | undefined {
  if (parseNode) {
    return parseNode.getNumberValue() || parseNode.getStringValue();
  }
  return undefined;
}
```